core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/tenant/JDBCTenantManager.java

```diff
@@ -903,21 +903,24 @@ public void deleteTenant(String tenantUniqueID) throws UserStoreException {
             clearTenantCaches(tenant);
             invalidateCacheManager(tenant.getDomain());
         }
-        String sqlStmt = TenantConstants.DELETE_TENANT_BY_UUID_SQL;
-        try (Connection dbConnection = getDBConnection()) {
-            try (PreparedStatement prepStmt = dbConnection.prepareStatement(sqlStmt)) {
-                prepStmt.setString(1, tenantUniqueID);
-                prepStmt.executeUpdate();
-                dbConnection.commit();
-            } catch (SQLException e) {
-                DatabaseUtil.rollBack(dbConnection);
+        Connection dbConnection = null;
+        try {
+            dbConnection = getDBConnection();
+            deleteUMTenantByTenantUUID(tenantUniqueID, dbConnection);
+            if (tenant != null) {
```

Comment:
I think it's better we throw an exception if the tenant is null. If we cannot get a tenant for the given UUID, that means that's an invalid UUID. And we need to do this at the beginning of the method.

Reply:
fixed 775da9a1e009be12698bc9ac9c86a8491b975b16